include/libtorrent/sha1_hash.hpp

```diff
@@ -49,66 +49,79 @@ POSSIBILITY OF SUCH DAMAGE.
 
 namespace libtorrent
 {
+	// TODO: find a better place for these functions
+	namespace aux
+	{
+		TORRENT_EXTRA_EXPORT void bits_shift_left(span<std::uint32_t> number, int n);
+		TORRENT_EXTRA_EXPORT void bits_shift_right(span<std::uint32_t> number, int n);
+	}
 
-	// This type holds a SHA-1 digest or any other kind of 20 byte
+	// This type holds an N digest or any other kind of N bits
 	// sequence. It implements a number of convenience functions, such
 	// as bit operations, comparison operators etc.
 	//
-	// In libtorrent it is primarily used to hold info-hashes, piece-hashes,
-	// peer IDs, node IDs etc.
-	class TORRENT_EXPORT sha1_hash
+	// This data structure is 32 bits aligned, like it's the case for
+	// each SHA-N specification.
+	template <int N>
+	class digest32
```

Comment:
at first I did it in another header, but I wanted to make the PR easy to read. This way is better to read how the `sha1_hash` was converted to a template